Regression in v0.28.1 (nil pointer dereference)

<!-- Please fill out the following questions, thanks! -->

Watchman Version: v0.28.1

**What were you trying to do?**

`docker run moov/watchman:v0.28.1`

**What did you expect to see?**

working container

**What did you see?**

```
ts=2024-03-12T20:32:32Z msg="Starting watchman server version v0.28.1" level=info
ts=2024-03-12T20:32:32Z msg="looking for  database provider" level=info
ts=2024-03-12T20:32:32Z msg="sqlite version 3.44.0" database=sqlite level=info
ts=2024-03-12T20:32:32Z msg="allowing only 1024 workers for search" level=info
ts=2024-03-12T20:32:32Z msg="Starting refresh of data" component=pipeline level=info
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x94ef55]

goroutine 1 [running]:
main.(*searcher).refreshData(0xc0001e7d40, {0x0, 0x0})
	/go/src/github.com/moov-io/watchman/cmd/server/download.go:277 +0x4b5
main.main()
	/go/src/github.com/moov-io/watchman/cmd/server/main.go:160 +0xf34

```

**How can we reproduce the problem?**

compare the output of  `docker run moov/watchman:v0.28.1`
`docker run moov/watchman:v0.28.0` (which is working version)

note: docker is not related to the issue, it is just a quick way to reproduce.

I have understood the problem already (quick fix) and will post a PR immediately

It is however completely strange to me why all the CI checks have passed, and the released version does not run any more...

--- a/cmd/server/download.go
+++ b/cmd/server/download.go
@@ -273,6 +273,9 @@ func (s *searcher) refreshData(initialDir string) (*DownloadStats, error) {
 		lastDataRefreshFailure.WithLabelValues("SDNs").Set(float64(time.Now().Unix()))
 		stats.Errors = append(stats.Errors, fmt.Errorf("OFAC: %v", err))
 	}
+	if results == nil {
+		results = &ofac.Results{}
+	}
 
 	sdns := precomputeSDNs(results.SDNs, results.Addresses, s.pipe)
 	adds := precomputeAddresses(results.Addresses)

--- a/pkg/download/client.go
+++ b/pkg/download/client.go
@@ -60,10 +60,8 @@ func (dl *Downloader) GetFiles(dir string, namesAndSources map[string]string) (m
 	}
 
 	// Check the initial directory for files we don't need to download
-	localFiles, err := os.ReadDir(dir)
-	if err != nil {
-		return nil, fmt.Errorf("readdir %s: %v", dir, err)
-	}
+	// do not treat an nonexisting directory as error
+	localFiles, _ := os.ReadDir(dir)
 
 	var mu sync.Mutex
 	out := make(map[string]io.ReadCloser)

